pandas/core/arrays/datetimes.py

```diff
@@ -1568,6 +1573,8 @@ def sequence_to_dt64ns(data, dtype=None, copy=False,
         copy = False
     elif isinstance(data, ABCSeries):
         data = data._values
+    elif isinstance(data, ABCPandasArray):
```

Comment:
Do you know if this is necessary? Why not fall through to the usual sequence handling?

Reply:
there was a test in tests.series.test_missing that ended up calling lib.infer_dtype and raising an AttributeError because PandasDType doesn't have a `base` attribute

Reply:
i seem to recall something about extension dtypes being registered in such a way that infer_dtype could pick up on them.  will take a look

Reply:
@jbrockmendel 
`infer_dtype` does not know about extension dtypes per se (and *cannot* because the point is to figure out whats hiding behind a dtype (mostly `object`), so dispatching based on dtype is not possible).

The only ones that are directly implemented are `Period` and `Interval`, but no other EA type, including `Datetime64TZ`.